packages/web-components/src/components/ic-step/ic-step.tsx

```diff
@@ -35,17 +59,102 @@ export class Step {
    */
   @Prop() lastStep!: boolean;
 
-  componentDidLoad(): void {
-    onComponentRequiredPropUndefined(
-      [
-        { prop: this.stepNum, propName: "step-num" },
-        { prop: this.lastStep, propName: "last-step" },
-      ],
-      "Step"
-    );
-  }
+  /**
+   * @internal The step number of the final step, managed by ic-stepper
+   */
+  @Prop() lastStepNum?: number;
+
+  /**
+   * @internal The name of the next step, managed by ic-stepper
+   */
+  @Prop() nextStepTitle?: string;
+
+  /**
+   * @internal The progress of the next step, calculated by dividing the current step number by the total number of steps, managed by ic-stepper
+   */
+  @Prop() progress?: number;
 
   render() {
+    // ARIA LABEL
+    let ariaLabel = "";
+
+    if (this.stepStatus === "complete" || this.stepType === "completed") {
+      ariaLabel = ". Completed step";
+    } else if (this.stepStatus === "not required") {
+      ariaLabel = ". Non-required step";
+    } else if (this.stepType === "disabled") {
+      ariaLabel = ". Disabled step";
+    }
+
+    // STEP STATUS FOR COMPACT STEP
+    let stepStatus;
+    if (this.stepStatus) {
+      stepStatus = this.stepStatus[0].toUpperCase() + this.stepStatus.slice(1);
+    }
+
+    // STATUS ICON FOR COMPACT STEP
+    let statusIcon;
+    if (this.stepStatus === "complete") {
+      statusIcon = (
+        <span
+          class="check-icon step-icon"
+          aria-hidden="true"
+          innerHTML={checkIcon}
+        ></span>
+      );
+    } else if (this.stepStatus === "not required") {
+      statusIcon = (
+        <span
+          class="warning-icon step-icon"
+          aria-hidden="true"
+          innerHTML={warningIcon}
+        ></span>
+      );
+    }
+
+    // COMPACT STEP COMPONENT
+    const compactStep = (
+      <div
+        class={{
+          ["step"]: true,
+          [`${this.stepStatus}`]: this.stepStatus !== "not required",
+          [`not-required`]: this.stepStatus === "not required",
+          ["current"]: this.current || this.stepType === "current",
+        }}
+      >
+        <ic-loading-indicator
```

Comment:
When using a screen reader (reading out everything on the page using Caps Lock + A), it reads out information about the loading indicator e.g. "Loading, 100%, progress indicator" before reading out the rest of the step information. 

Do you think we should set aria-hidden or something on the loading indicator so the screen reader doesn't pick it up? Don't think it needs to be read out at all as it's decorative and might cause confusion with the reference to loading, plus all the information is read out from the actual step